hello-app-tls/main.go

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// [START all]
+package main
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+)
+
+func main() {
+	// use PORT environment variable, or default to 8443
+	port := "8443"
+	if fromEnv := os.Getenv("PORT"); fromEnv != "" {
+		port = fromEnv
+	}
+
+	tlsCert, tlsKey := os.Getenv("TLS_CERT"), os.Getenv("TLS_KEY")
+	if tlsCert == "" {
+		log.Fatal("TLS_CERT environment variable not set")
+	}
+	if tlsKey == "" {
+		log.Fatal("TLS_KEY environment variable not set")
+	}
+
+	// register hello function to handle all requests
+	server := http.NewServeMux()
+	server.HandleFunc("/", hello)
+
+	// start the web server on port and accept requests
+	log.Printf("tls cert: %s", tlsCert)
+	log.Printf("tls key: %s", tlsKey)
+	log.Printf("Server listening on port %s", port)
+	err := http.ListenAndServeTLS(":"+port, tlsCert, tlsKey, server)
```

Comment:
A common way of writing this is:

```
if err := http.ListenAndServeTLS(":"+port, tlsCert, tlsKey, server); err != nil {
    log.Fatal(err)
}
```

Reply:
that was confusing to many readers since this is not meant to be a golang sample.

Reply:
`log.Fatal(http.ListenAndServe...)` is idiomatic.

Reply:
I'm aware, but we actually got multiple tutorial feedback indicating the readers from other languages don't understand what's going on